project/src/test/java/com/google/sps/EventSpannerTasksTest.java

```diff
@@ -116,36 +128,68 @@ public void getEventsByIdTest() {
+    Mockito.when(request.getParameter("date")).thenReturn(DATE_STRING);
+    Mockito.when(request.getParameter("time")).thenReturn(TIME);
+    Mockito.when(request.getParameter("description")).thenReturn(DESCRIPTION);
+    Mockito.when(request.getParameter("location")).thenReturn(LOCATION);
+    Mockito.when(request.getParameter("interests")).thenReturn("Tech, Work");
 
     new EventCreationServlet().doPost(request, response);
+    Event event =
+        new Event.Builder(EVENT_NAME, DESCRIPTION, LABELS, LOCATION, DATE, TIME, HOST).setId(stringWriter.toString().trim().split("\"")[3]).build();
 
-    // Get back Event posted in db
-    Event returnedEvent = new Gson().fromJson(response.getContentAsString(), Event.class);
-
-    // Check that event inserted into doPost is same as event returned from doPost db
-    Assert.assertEquals(returnedEvent.getName(), EVENT_NAME);
-    Assert.assertEquals(returnedEvent.getDescription(), DESCRIPTION);
-    Assert.assertEquals(returnedEvent.getLocation(), LOCATION);
-    Assert.assertEquals(returnedEvent.getDate(), DATE);
-    Assert.assertEquals(returnedEvent.getTime(), TIME);
-    Assert.assertEquals(returnedEvent.getHost().getName(), HOST_NAME);
-    Assert.assertEquals(returnedEvent.getHost().getEmail(), EMAIL);
+    try {
+    JSONAssert.assertEquals(CommonUtils.convertToJson(event).trim(), stringWriter.toString().trim(), /*assert order= */ false);
+    } catch (JSONException e) {
+        System.out.println("JSON conversion failed.");
+    }
   }
 
+  /**
+   * Verify putting event in database through /create-event doPost with correct params and
+   * getting back correct event out of db with NLP API labeling suggestions
+   */
+  @Test
+  public void testEventCreationDoPost_Nlp() throws Exception {
+    SpannerTasks.insertOrUpdateUser(HOST);
+    setAuthenticationHelper();
+    Mockito.when(request.getParameter("name")).thenReturn(EVENT_NAME);
+    Mockito.when(request.getParameter("date")).thenReturn(DATE_STRING);
+    Mockito.when(request.getParameter("time")).thenReturn(TIME);
+    String DESCRIPTION_COOKING_CLASS =
+    "Come learn to meal prep with us. This class will include prepping foods beyond sandwiches or salads."
+    + " We will learn to bake desserts as well. Bring a friend!";
+
+    Mockito.when(request.getParameter("description")).thenReturn(DESCRIPTION_COOKING_CLASS);
+    Mockito.when(request.getParameter("location")).thenReturn(LOCATION);
+    Mockito.when(request.getParameter("interests")).thenReturn("Cooking");
+    String text = new StringBuilder().append(EVENT_NAME).append(" ").append(DESCRIPTION_COOKING_CLASS).toString();
+
+    //Mock NLP API response with real category response
+    EventCreationServlet servlet = Mockito.spy(EventCreationServlet.class);
+    Mockito.doReturn(new ArrayList<>(Arrays.asList("Jobs and Education", "Food and Drink"))).when(servlet).getNlpSuggestedFilters(text, new ArrayList<String>());
```

Comment:
Since you are testing the servlet. I don't think you should mock a function of it. I think what you want to mock is the return of the NLP API.

Reply:
done, thank you